### PR TITLE
Add forward connector

### PR DESCRIPTION
**Pull request by djaglowski**

### CODE REVIEW

**Comment by djaglowski** [comment]
cc @bogdandrutu

File: connector/forwardconnector/forward.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+// This is the map of already created forward connectors for particular configurations.
+// We maintain this map because the Factory is asked trace, metric, and log receivers
+// separately but they must not create separate objects. When the connector is shutdown
+// it should be removed from this map so the same configuration can be recreated successfully.
+var connectors = sharedcomponent.NewSharedComponents()
+
+// forwardConnector is used to pass signals directly from one pipeline to another.
+// This is useful when there is a need to replicate data and process it in more
+// than one way. It can also be used to join pipelines together.
+type forwardConnector struct {
```
Comment (bogdandrutu) [inline connector/forwardconnector/forward.go:106]: s/forwardConnector/forward?

File: connector/forwardconnector/forward.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+	return consumer.Capabilities{MutatesData: false}
+}
+
+func (c *forwardConnector) Start(_ context.Context, host component.Host) error {
+	return nil
+}
+
+func (c *forwardConnector) Shutdown(ctx context.Context) error {
+	return nil
+}
```
Comment (bogdandrutu) [inline connector/forwardconnector/forward.go:122]: Include the `component.StartFunc` and `component.ShutdownFunc` instead.

**Review by bogdandrutu — APPROVED** [review]

File: connector/forwardconnector/forward.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+	conn := comp.Unwrap().(*forwardConnector)
+	conn.Logs = nextConsumer
+	return conn, nil
+}
+
+// This is the map of already created forward connectors for particular configurations.
+// We maintain this map because the Factory is asked trace, metric, and log receivers
+// separately but they must not create separate objects. When the connector is shutdown
+// it should be removed from this map so the same configuration can be recreated successfully.
+var connectors = sharedcomponent.NewSharedComponents()
```
Comment (bogdandrutu) [inline connector/forwardconnector/forward.go:101]: I know I am using it as global in other receiver. I think we can use this as member of the Factory.

File: connector/forwardconnector/forward.go (hunk 1 above)
Comment (djaglowski) [reply connector/forwardconnector/forward.go]: Fixed

File: connector/forwardconnector/forward.go (hunk 2 above)
Comment (djaglowski) [reply connector/forwardconnector/forward.go]: Done

File: connector/forwardconnector/forward.go [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+	return c.Traces.ConsumeTraces(ctx, td)
+}
+
+func (c *forward) ConsumeMetrics(ctx context.Context, md pmetric.Metrics) error {
+	return c.Metrics.ConsumeMetrics(ctx, md)
+}
+
+func (c *forward) ConsumeLogs(ctx context.Context, ld plog.Logs) error {
+	return c.Logs.ConsumeLogs(ctx, ld)
+}
```
Comment (bogdandrutu) [inline connector/forwardconnector/forward.go:131]: If I don't miss something, you don't need these 3 func overrides.

Comment (djaglowski) [reply connector/forwardconnector/forward.go]: Good point. This was an artifact of the earlier implementation (e.g. `ConsumerLogsToLogs`). Removed.

**Pull request merged**